packages/neon_framework/packages/neon_http_client/lib/src/neon_http_client.dart

```diff
@@ -97,16 +102,9 @@ final class NeonHttpClient with http.BaseClient {
     var interceptedRequest = request;
     for (final interceptor in interceptors) {
       if (interceptor.shouldInterceptRequest(interceptedRequest)) {
-        try {
-          interceptedRequest = await interceptor.interceptRequest(
-            request: interceptedRequest,
-          );
-        } catch (_, stackTrace) {
-          Error.throwWithStackTrace(
-            InterceptionException('Failed to intercept request', request.url),
-            stackTrace,
-          );
-        }
+        interceptedRequest = await interceptor.interceptRequest(
+          request: interceptedRequest,
+        );
       }
```

Comment:
I'm not a fan of removing the try catch.
An http client should wrap all errors into `http.ClientException`s. In the unlikely case, a Interceptor trows I still want the exceptions to be wrapped as we only catch `http.ClientException`s down the road (see the request manager).

We either need to find a different solution or just rethrow the exception un-wrapped.